aws/data_source_aws_ssoadmin_role.go

```diff
@@ -0,0 +1,130 @@
+package aws
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func dataSourceAwsSsoAdminRole() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsSsoAdminRoleRead,
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"assume_role_policy": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"path": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"permissions_boundary": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"unique_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"permission_set_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"create_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"max_session_duration": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"tags": tagsSchemaComputed(),
+		},
+	}
+}
+
+func dataSourceAwsSsoAdminRoleRead(d *schema.ResourceData, meta interface{}) error {
+	iamconn := meta.(*AWSClient).iamconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	permissionSetName := d.Get("permission_set_name").(string)
+	pathPrefix := aws.String("/aws-reserved/sso.amazonaws.com/")
+	roleNamePrefix := fmt.Sprintf("AWSReservedSSO_%s_", permissionSetName)
+
+	// Get all AWS SSO roles
+	input := &iam.ListRolesInput{
+		PathPrefix: pathPrefix,
+	}
+
+	roles := []*iam.Role{}
+
+	err := iamconn.ListRolesPages(
+		input,
+		func(page *iam.ListRolesOutput, lastPage bool) bool {
+			for _, role := range page.Roles {
+				if strings.HasPrefix(aws.StringValue(role.RoleName), roleNamePrefix) {
+					roles = append(roles, role)
+				}
+			}
+			return !lastPage
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("error reading roles with path prefix (%s): %w", aws.StringValue(pathPrefix), err)
+	}
+
+	if len(roles) > 1 {
```

Comment:
I'm guessing this could happen if there are multiple AWS accounts which are federating access to a single tenant account using the same Permission Set name? Maybe that's why the IAM roles auto-generated by AWS are suffixed with some hash. Might be worth adding a note to the docs about this edge case, if indeed it can happen.

Reply:
AWS SSO doesn't create a role for every assignment for the same permission set. That's actually what the `TestAccAwsDataSourceSsoAdminRole_UserAndGroup` acceptance test is testing. The length check is here as a safety precaution but is probably not needed.